core/src/main/java/org/apache/calcite/rel/rel2sql/RelToSqlConverter.java

```diff
@@ -975,7 +975,14 @@ protected void buildAggGroupList(Aggregate e, Builder builder,
   protected Builder buildAggregate(Aggregate e, Builder builder,
       List<SqlNode> selectList, List<SqlNode> groupByList) {
     for (AggregateCall aggCall : e.getAggCallList()) {
-      SqlNode aggCallSqlNode = builder.context.toSql(aggCall);
+      String delimiter = null;
+      if ((aggCall.getAggregation().kind == SqlKind.LISTAGG
+          || aggCall.getAggregation().kind == SqlKind.STRING_AGG) && aggCall.getArgList().size() == 2
+          && e.getInput() instanceof Project) {
+        delimiter =
+            ((Project) e.getInput()).getChildExps().get(aggCall.getArgList().get(1)).toString();
```

Comment:
here,  if we are sure about the delimiter would be RexLiteral, fetch the call the `getValue2().toString()`